pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -220,6 +222,23 @@ func Convert_v1_Disk_To_api_Disk(c *ConverterContext, diskDevice *v1.Disk, disk
 		disk.BootOrder = &api.BootOrder{Order: *diskDevice.BootOrder}
 	}
 
+	if diskDevice.BlockSize != nil {
+		if blockSize := diskDevice.BlockSize.CustomBlockSize; blockSize != nil {
+			disk.BlockIO = &api.BlockIO{
+				LogicalBlockSize:  blockSize.LogicalBlockSize,
+				PhysicalBlockSize: blockSize.PhysicalBlockSize,
+			}
+		} else if matchFeature := diskDevice.BlockSize.MatchVolume; matchFeature != nil && matchFeature.Enabled != nil && *matchFeature.Enabled {
```

Comment:
The FeatureState struct is deceiving. A non-null featureState defaults to "true"

so setting ```matchFeature:{}``` in the yaml is the same as
```
matchFeature:
    enabled: true
```

This means `diskDevice.BlockSize.MatchVolume.Enabled == nil` is the same as `*diskDevice.BlockSize.MatchVolume.Enabled == true`


False would be a null MatchVolume or *MatchVolume.Enabled = false. 

more info about FeatureStates below. there's a history for why this is the way it is. it has to do with our VMIPresets and field merging/overriding. it's annoying. 

```
// Represents if a feature is enabled or disabled.
//
// +k8s:openapi-gen=true
type FeatureState struct {
        // Enabled determines if the feature should be enabled or disabled on the guest.
        // Defaults to true.
        // +optional
        Enabled *bool `json:"enabled,omitempty"`
}

```

Reply:
Thanks for the heads-up. That is indeed not what I expected.